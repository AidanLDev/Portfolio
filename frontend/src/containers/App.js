import React, { useState } from "react";
import { Route, Switch, Redirect } from "react-router-dom";
import { mdiMenu } from "@mdi/js";
import Icon from "@mdi/react";

import Navigation from "containers/Navigation/Navigation";
import Portfolio from "pages/Portfolio/Portfolio";
import Projects from "containers/Projects/Projects";
import Footer from "containers/Footer/Footer";

import styles from "./Style";
import Sidedrawer from "./Navigation/Sidedrawer/Sidedrawer";

const App = () => {
  const [showSideDrawer, setShowSideDrawer] = useState(false);

  const routes = (
    <Switch>
      <Route path="/" exact component={Portfolio} />
      <Route path="/projects" component={Projects} />
      <Redirect to="/" />
    </Switch>
  );

  const closeSideDrawer = () => setShowSideDrawer(false);
  const openSideDrawer = () => setShowSideDrawer(true);

  return (
    <div className={styles.App}>
      <header className={styles.AppHeader}>
        <nav className={styles.desktopOnly}>
          <Navigation />
        </nav>
        <nav>
          <div className={styles.mobileOnly}>
            <Icon
              path={mdiMenu}
              size={2}
              color={"white"}
              onClick={openSideDrawer}
            />
            {/* Logo */}
          </div>
          {showSideDrawer && (
            <Sidedrawer
              showDrawer={showSideDrawer}
              closeDrawer={closeSideDrawer}
            />
          )}
        </nav>
      </header>
      <div className={styles.MainContent}>{routes}</div>
      <Footer />
    </div>
  );
};

export default App;
