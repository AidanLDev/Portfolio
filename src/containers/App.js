import React from 'react'
import { Route, Switch, withRouter, Redirect } from 'react-router-dom'

import Navigation from 'containers/Navigation/Navigation'
import Achievements from 'containers/Achievements/Achievements'
import Projects from 'containers/Projects/Projects'

import styles from './Style'

const App = () => {

  const routes = (
    <Switch>
      <Route path="/" exact component={Projects} />
      <Route path="/achievements" component={Achievements} />
      <Redirect to="/" />
    </Switch>
  )

  return (
    <div className={styles.App}>
      <header className={styles.AppHeader}>
        <Navigation />
      </header>
      <div className={styles.MainContent}>
        {routes}
      </div>
    </div>
  );
}

export default App;
