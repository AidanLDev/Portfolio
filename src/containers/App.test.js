import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

it('Renders App Component Correctly', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
