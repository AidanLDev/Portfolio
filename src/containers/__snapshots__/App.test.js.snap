// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App Component Correctly 1`] = `
<div
  className="App"
>
  <header
    className="AppHeader"
  >
    <nav
      className="desktopOnly"
    >
      <Navigation />
    </nav>
    <nav>
      <div
        className="mobileOnly"
        onClick={[Function]}
      >
        <Icon
          color="white"
          horizontal={false}
          path="M3,6H21V8H3V6M3,11H21V13H3V11M3,16H21V18H3V16Z"
          rotate={0}
          size={2}
          spin={false}
          vertical={false}
        />
      </div>
    </nav>
  </header>
  <div
    className="MainContent"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/blog/search"
      />
      <Route
        component={[Function]}
        path="/blog"
      />
      <Route
        component={[Function]}
        path="/map"
      />
      <Redirect
        to="/"
      />
    </Switch>
  </div>
  <Footer />
</div>
`;
